### ISSUE TITLE
Feature request: add onOpen onClosed events

### ISSUE THREAD

**Issue by arifje** [headmandev/vue3-side-panel#5]
Nice component, would like to see some events for opening/closing! 🥳

**Comment by headmandev** [comment]
@arifje 

There are no events due to repetition of `v-model` change logic. 
Under the hood the close action is 

```js
const closePanel = () => emit('update:modelValue', false);
```

This means that we can watch the changes with `isOpened`
(In this case, you can catch the opening and closing event yourself)

```vue
<script setup>
import { ref, watch } from 'vue';
import { VueSidePanel } from 'vue3-side-panel';

const isOpened = ref(false);
watch(isOpened, (newState, oldState) => {
  if (newState) return actionAfterModalOpening();
  
  // otherwise the modal has been closed
  actionAfterModalClosing();
})

</script>

<template>
<VueSidePanel v-model="isOpened">
  <div style="height: 100%; background-color: #ccc">
    <h2> This is scrolled body! </h2>
  </div>    
</VueSidePanel>
</template>

```

There's only one thing to worry about: when you close the modal window, the isOpened variable changes, but the closing animation still works for a while.

So adding a close event to `VueSidePanel` to know when the modal is completely closed and the animation is complete makes sense.
This could be a feature request - I agree, but I think in most cases just `watch(isOpened, (newState, oldState) => {` is enough.

**Comment by headmandev** [comment]
@arifje 
`@opened` and `@closed` events are now in version [1.3.0](https://github.com/headmandev/vue3-side-panel/pull/6)
